src/databricks/labs/remorph/reconcile/query_builder/base.py

```diff
@@ -162,3 +178,48 @@ def _get_custom_transformation(
             transform_rule_mapping.append(TransformRuleMapping(col_origin, transform, col_alias))
 
         return transform_rule_mapping
+
+    @staticmethod
+    def _get_threshold_select_function(match_type, mode):
+        try:
+            switch = {
+                (match_type, mode): ThresholdSQLTemplate.SELECT_NUMBER_ABSOLUTE.value
+                for match_type in Constants.NUMBER_TYPES
+                if mode == ThresholdMode.ABSOLUTE.value
+            }
+            switch.update(
+                {
+                    (match_type, mode): ThresholdSQLTemplate.SELECT_NUMBER_PERCENTILE.value
+                    for match_type in Constants.NUMBER_TYPES
+                    if mode == ThresholdMode.PERCENTILE.value
+                }
+            )
+            switch.update(
+                {
+                    (match_type, mode): ThresholdSQLTemplate.SELECT_DATETIME.value
+                    for match_type in Constants.DATETIME_TYPES
+                    if mode == ThresholdMode.ABSOLUTE.value
+                }
+            )
+
+            return switch[(match_type, mode)]
+        except KeyError as exc:
+            error_message = f"Invalid threshold match type or mode in get_threshold_select_function: {match_type} or mode: {mode} error -> {exc!s}"
+            logger.error(error_message)
+            raise ValueError(error_message) from exc
+
+    @staticmethod
+    def _get_threshold_filter_function(match_type):
+        try:
```

Comment:
Same remark as above: nested if/else would be more readable

Reply:
Done